x-pack/plugins/infra/public/containers/logs/log_position/use_log_position.ts

```diff
@@ -146,24 +90,63 @@ export const useLogPositionState: () => LogPositionStateParams & LogPositionCall
     RELATIVE_END_UPDATE_DELAY
   );
   useEffect(() => {
-    if (dateRange.endDateExpression !== 'now') {
+    if (timeRange.to !== 'now') {
       return;
     }
 
     // User is close to the bottom edge of the scroll.
     if (throttledPagesAfterEnd <= DESIRED_BUFFER_PAGES) {
-      logPositionStateContainer.transitions.updateTimeRange({ to: 'now' });
+      logStreamPageSend({
+        type: 'UPDATE_TIME_RANGE',
+        timeRange: { to: 'now' },
+      });
     }
-  }, [dateRange.endDateExpression, throttledPagesAfterEnd, logPositionStateContainer]);
-
-  useInterval(
-    () => logPositionStateContainer.transitions.updateTimeRange({ from: 'now-1d', to: 'now' }),
-    latestLogPositionState.refreshInterval.pause ||
-      latestLogPositionState.refreshInterval.value <= 0
-      ? null
-      : latestLogPositionState.refreshInterval.value
+  }, [timeRange.to, throttledPagesAfterEnd, logStreamPageSend]);
+
+  const jumpToTargetPosition = useCallback(
```

Comment:
**suggestion(non-blocking)**: just noticed these 5 actions all depend on the `logStreamPageSend` function.
Using `useCallback` is correct to memorize these actions, and on each re-render react will do 5 times the necessary work to retrieve their memo references.
A small optimization could be to return these actions into an object and memoize it using `useMemo`, which will produce the same result invoking one single hook:
```ts
const actions = useMemo(
  () => ({
    jumpToTargetPosition: (_targetPosition: TimeKey | null) => {
      logStreamPageSend({ type: 'JUMP_TO_TARGET_POSITION', targetPosition: _targetPosition });
    },
    jumpToTargetPositionTime: (time: number) => {
      logStreamPageSend({ type: 'JUMP_TO_TARGET_POSITION', targetPosition: { time } });
    },
    reportVisiblePositions: (_visiblePositions: VisiblePositions) => {
      logStreamPageSend({ type: 'REPORT_VISIBLE_POSITIONS', visiblePositions: _visiblePositions });
    },
    startLiveStreaming: () => {
      logStreamPageSend({ type: 'UPDATE_REFRESH_INTERVAL', refreshInterval: { pause: false } });
    },
    stopLiveStreaming: () => {
      logStreamPageSend({ type: 'UPDATE_REFRESH_INTERVAL', refreshInterval: { pause: true } });
    },
  }),
  [logStreamPageSend]
);
```

then you can either destructure each action from the returned object or directly spread it into the returned object.

Reply:
I think the performance difference will be negligible, but since it's an easy change this looks like a good suggestion to me 👍